### PR TITLE
MULE-10223: Move MuleEventContext to compatibility module

### PR DESCRIPTION
**Pull request by elrodro83**
* First pass

### CODE REVIEW

File: tests/component-plugin/src/main/java/org/mule/functional/api/component/FunctionalTestComponent.java [hunk 1]
```diff
@@ -392,14 +391,6 @@ public void setWaitTime(long waitTime) {
     this.waitTime = waitTime;
   }
```
Comment (pablokraan) [inline tests/component-plugin/src/main/java/org/mule/functional/api/component/FunctionalTestComponent.java]: Maybe FunctionalTestComponent can be renamed to FunctionalTestProcessor

File: tests/component-plugin/src/main/resources/META-INF/mule-test.xsd [hunk 2]
```diff
@@ -108,7 +101,7 @@
                 <xsd:attribute name="appendString" type="xsd:string">
                     <xsd:annotation>
                         <xsd:documentation>
-                            A string value that will be appeneded to every message payload that passes through the component. Note that by setting this property you implicitly select that the message payload will be converted to a string and that a string payload will be returned. The inbound transformer (if any) will get applied first, but if that does not return a string, {{MuleEventContext.getMessageAsString()}} will be called directly after.
+                            A string value that will be appeneded to every message payload that passes through the component. Note that by setting this property you implicitly select that the message payload will be converted to a string and that a string payload will be returned.
```
Comment (pablokraan) [inline tests/component-plugin/src/main/resources/META-INF/mule-test.xsd]: appeneded > appended

Comment (pablokraan) [inline tests/component-plugin/src/main/resources/META-INF/mule-test.xsd]: component > processor?

**Comment by elrodro83** [comment]
Igual think Custom processor belongs to the test plugin rather than compatibility. Otherwise, Wen should have a test plugin forma each module that uses custom-processor to migrate yo.

**Pull request merged**